bin/nuxt-dev

```diff
@@ -59,14 +60,17 @@ defaultsDeep(nuxtConfig, { watchers: { chokidar: { ignoreInitial: true } } })
 let dev = startDev()
 let needToRestart = false
 
-// Start watching for nuxt.config.js changes
+// Start watching for nuxt.config.js and user-defined changes
 chokidar
-  .watch(nuxtConfigFile(argv), nuxtConfig.watchers.chokidar)
-  .on('all', () => {
-    consola.debug('[nuxt.config.js] changed')
+  .watch((nuxtConfig.watch || []).concat(nuxtConfigFile(argv)), nuxtConfig.watchers.chokidar)
```

Comment:
I think it will be better if move `watch` config into `watchers` and also set `nuxtConfigFile` as default value of `watch`, how do you think @galvez ?

Reply:
@clarkdo good call -- however I was talking with @Atinux and he came up with the idea of working out a way to isolate that funcionality in `lib/builder/builder.js` (`fileWatchers`), so the same functionality is available in Nuxt's programmatic usage. I'm working on it, let's see if I can get to a working prototype soon :)